src/components/Articles.jsx

```diff
@@ -1,29 +1,36 @@
 import { useEffect, useState } from 'react';
-import { getArticles } from '../utils/api';
+import { Link, useSearchParams } from 'react-router-dom';
+import { getArticles, getTopics } from '../utils/api';
+import SubNav from './SubNav';
 import Loader from './Loader';
 import './Articles.css';
-import { Link } from 'react-router-dom';
 
 function Articles() {
   const [articles, setArticles] = useState([]);
+  const [topics, setTopics] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const topic = searchParams.get('topic') || undefined;
```

Comment:
Instead of undefined here I would suggest going with an empty string which should also achieve the same result.